Sources/Navigation/DistributedNavigationDelegate.swift

```diff
@@ -539,32 +539,51 @@ extension DistributedNavigationDelegate: WKNavigationDelegate {
 
     @MainActor
     public func webView(_ webView: WKWebView, didStartProvisionalNavigation wkNavigation: WKNavigation?) {
-        let navigation: Navigation
-        if let expectedNavigation = navigationExpectedToStart,
-           wkNavigation != nil
-            || expectedNavigation.navigationAction.navigationType == .sessionRestoration
-            || expectedNavigation.navigationAction.url.scheme.map(URL.NavigationalScheme.init) == .about {
+        var navigation: Navigation
+        if let approvedNavigation = wkNavigation?.navigation,
+           approvedNavigation.state == .approved, approvedNavigation.navigationActions.isEmpty == false {
+            // rely on the associated Navigation that is in the correct state
+            navigation = approvedNavigation
+
+        } else if let expectedNavigation = navigationExpectedToStart,
+                  wkNavigation != nil
+                    || expectedNavigation.navigationAction.navigationType == .sessionRestoration
+                    || expectedNavigation.navigationAction.url.scheme.map(URL.NavigationalScheme.init) == .about {
```

Comment:
Just a suggestion to make the code more readable. If these conditions are defined before the `if` as Bool values, their names can describe the purpose and make it easy to process for readers.

`let isUrlSchemeAbout = ...`

`if isUrlSchemeAbout || is...`